### ISSUE TITLE
Data reading clarifications

### ISSUE THREAD

**Issue by rwegener2** [ICESat2-SlideRule/h5coro#8]
This is a bit more of a discussion than an issue, but I thought I'd start documenting. Here are a few things I've noticed while using h5coro:

### 1 - `readDatasets()` `datasets` parameter
The `.readDatasets()` method wants a `datasets` parameter of either a List or Dictionary (I initially gave it a string). I don't think this is something to change, but once we think the project is stable enough we might consider either docs or implementing some [typing](https://docs.python.org/3/library/typing.html). Either or both of those would make debugging easier.

### 2 - Multiple `readDatasets` calls
Asking for the same dataset twice returns `UnboundLocalError: cannot access local variable 'dataset_level' where it is not associated with a value`. Again, not the most pressing issue, but it came up for me because I first tried to run something like `quality_ph = h5obj.readDatasets(datasets=['gt1r/heights/quality_ph'])`, then when I retried with just `h5obj.readDatasets(datasets=['gt1r/heights/quality_ph'])` I got the error. At some point in the future it could be nice to make that request idempotent.

### 3 - Need to include '/'?

Just to clarify, it seems that you should be able to either use the `/` at the beginning of a path or omit it, and both should still reference starting at the root node. Is that correct? For example, all the following syntaxes should work:

```python
h5obj.readDatasets(datasets=['/gt1r/heights/quality_ph'])
h5obj.readDatasets(datasets=['gt1r/heights/quality_ph'])
h5obj.readAttribute('/gt1r/heights/data_rate')
h5obj.readAttribute('gt1r/heights/data_rate')
h5obj.readAttribute('/short_name')
h5obj.readAttribute('short_name')
```

### 4 - `inspectVariable` Errors

I'm running:
```python
from h5coro import h5coro, s3driver, filedriver

h5coro.config(errorChecking=True, verbose=False)

s3url_atl03 = 'nsidc-cumulus-prod-protected/ATLAS/ATL03/006/2019/11/30/ATL03_20191130112041_09860505_006_01.h5'
h5obj = h5coro.H5Coro(s3url_atl03, 
                      s3driver.S3Driver,
                      credentials=s3_creds_dict)

metadata, attrs = h5obj.inspectVariable('/gt1l/heights/lat_ph')
```
and I get:
```
H5Coro encountered an error processing nsidc-cumulus-prod-protected/ATLAS/ATL03/006/2019/11/30/ATL03_20191130112041_09860505_006_01.h5//gt1l/heights/lat_ph/DIMENSION_LIST: variable length data types require reading a global heap, which is not yet supported
H5Coro encountered an error processing nsidc-cumulus-prod-protected/ATLAS/ATL03/006/2019/11/30/ATL03_20191130112041_09860505_006_01.h5//gt1l/heights/lat_ph/valid_min: failed to read expected bytes for datatype message: 24 != 20
H5Coro encountered an error processing nsidc-cumulus-prod-protected/ATLAS/ATL03/006/2019/11/30/ATL03_20191130112041_09860505_006_01.h5//gt1l/heights/lat_ph/valid_max: failed to read expected bytes for datatype message: 24 != 20
```
the function returns all the other attributes fine and gives an empty list for the 3 listed above. Is this expected?

### Next Steps
I'm happy to help with development on any of these points. 1 and 2 are just "documenting these for future reference/collecting user experience". 3 and 4 are questions, and any insight is appreciated. Thanks!

**Comment by jpswinski** [comment]
@rwegener2 - thanks for enumerating these issues.  3 of the 4 are handled in the branch behind PR #5, so are you okay if I merge that in now?  

Specifically:
1 - sounds good to me!  I haven't used the typing feature of Python yet, but think it is a good idea
2 - the leading and trailing slashes should be optional and handled
3 - this is fixed in PR #5 
4 - this is fixed in PR #5

**Comment by jpswinski** [comment]
PR #5 has been merged
